rust/src/main.rs

```diff
@@ -73,17 +72,86 @@ fn main() -> anyhow::Result<()> {
         }
     }));
 
+    #[cfg(any(
+        target_os = "openbsd",
+        target_os = "netbsd",
+        target_os = "freebsd",
+        target_os = "dragonfly",
+        target_os = "macos",
+        target_os = "ios",
+        target_os = "watchos",
+    ))]
+    let enabled_flags: HashSet<_> = config
+        .features
+        .file_flags
+        .unwrap_or_default()
+        .iter()
+        .cloned()
+        .collect();
+
     for test_case in TEST_CASES {
-        if let Some(sc) = &test_case.syscall {
-            if !enabled_features.contains(sc) {
-                println!(
-                    "skipped {}: please add it to the configuration file if you want to run it",
-                    sc
-                );
-                continue;
+        //TODO: There's probably a better way to do this...
+        let mut should_skip = false;
+
+        let mut message = None;
+
+        let features = test_case
+            .required_features
+            .iter()
+            .cloned()
+            .collect::<HashSet<_>>();
+        let missing_features = features.difference(&enabled_features);
+        if missing_features.clone().count() > 0 {
```

Comment:
Why the `clone()`?

Reply:
Because otherwise the elements are references instead of being owned.